pilot/pkg/model/push_context.go

```diff
@@ -682,22 +682,30 @@ func (ps *PushContext) GatewayServices(proxy *Proxy) []*Service {
 
 // Services returns the list of services that are visible to a Proxy in a given config namespace
 func (ps *PushContext) Services(proxy *Proxy) []*Service {
+	if proxy != nil {
+		return ps.ServicesBySidecar(proxy.SidecarScope, proxy.ConfigNamespace)
+	}
+	return ps.ServicesBySidecar(nil, "")
+}
+
+// Services returns the list of services that are visible to a Proxy in a given config namespace
+func (ps *PushContext) ServicesBySidecar(sidecarScope *SidecarScope, configNamespace string) []*Service {
```

Comment:
I mean this kind of method could be called directly, which is confusing

Reply:
Do you mean by contructing proxy object before calling?

Reply:
Ok. Found a way to revert these changes